### ISSUE TITLE
[ENHANCEMENT] add debug parameter with elemental command

### ISSUE THREAD

**Issue by Vicente-Cheng** [harvester/harvester#4541]
**Is your enhancement related to a problem? Please describe.**
Sometimes, we will make errors when we use elemental commands like `elemental install` or `elemental upgrade`.
Adding `debug` parameter would help us to clarify the root cause.

**Describe the solution you'd like**
elemental command already provides the debug parameter. We could add it.

**Describe alternatives you've considered**
None

**Additional context**
Make sure the install and upgrade work as usual, and check the log to confirm that contains debug info.

**Comment by bk201** [comment]
@Vicente-Cheng We need a test plan and move it to the Ready For Testing stage.

**Comment by bk201** [comment]
Updated test plan in https://github.com/harvester/harvester/issues/4541#issuecomment-1727133957

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.2-04b7f9d7` (10/03). Close this issue.

### Result
- [x] 1. On the single node Harvester, we  can see elemental has `DEBUG` level logs (with level DEBU  in the lines) in the post-drain state to upgrade OS



    * During the upgrade process proceed to the `post-drain` state or after upgrade successfully complete 
    * When we check the upgrade related pods in harvester-system 
      ```
      $ kubectl get pod -n harvester-system
      
      harv112upd:~ # kubectl get pod -n harvester-system
      NAME                                                        READY   STATUS      RESTARTS      AGE
      harvester-7f86459fdb-92n27                                  1/1     Running     3 (43m ago)   54m
      harvester-load-balancer-6d89b964bb-6bgxb                    1/1     Running     2 (40m ago)   54m
      harvester-load-balancer-webhook-6dd77c56bf-dvsxc            1/1     Running     1 (40m ago)   54m
      harvester-network-controller-manager-68fd49b88f-5ch2g       1/1     Running     2 (40m ago)   54m
      harvester-network-controller-w6j5v                          1/1     Running     2 (39m ago)   54m
      harvester-network-webhook-697c754ffb-9k7qb                  1/1     Running     1 (40m ago)   54m
      harvester-node-disk-manager-lxfpd                           1/1     Running     2 (38m ago)   54m
      harvester-node-manager-9x2l5                                1/1     Running     1 (40m ago)   54m
      harvester-webhook-7867f89d66-bjxg9                          1/1     Running     1 (40m ago)   54m
      hvst-upgrade-sfgvp-apply-manifests-9fm7w                    0/1     Completed   0             58m
      hvst-upgrade-sfgvp-single-node-upgrade-harv112upd-j27b2     0/1     Completed   0             47m
      ```
    * Then on the single node upgrade log, we can find the logs in line already marked with the `DEBU` label
      ```
      $ kubectl logs hvst-upgrade-sfgvp-single-node-upgrade-harv112upd-j27b2 -n harvester-system
      
      DEBU[2023-10-04T09:27:15Z] Starting elemental version 0.3.1 on commit bab71d15 
      INFO[2023-10-04T09:27:15Z] reading configuration form '/tmp/tmp.Z4msHlfwTF' 
      DEBU[2023-10-04T09:27:15Z] Full config loaded: &v1.RunConfig{
      ...
      
      INFO[2023-10-04T09:27:15Z] Upgrade called                               
      DEBU[2023-10-04T09:27:15Z] Mounting partition COS_STATE                 
      DEBU[2023-10-04T09:27:15Z] Mounting partition COS_RECOVERY              
      INFO[2023-10-04T09:27:16Z] Applying 'before-upgrade' hook               
      INFO[2023-10-04T09:27:16Z] Running before-upgrade hook                  
      INFO[2023-10-04T09:27:16Z] deploying image /tmp/tmp.ttDBERFvDc to /run/initramfs/cos-state/cOS/transition.img 
      INFO[2023-10-04T09:27:16Z] Preparing root tree for image: /run/initramfs/cos-state/cOS/transition.img 
      INFO[2023-10-04T09:27:16Z] Copying /tmp/tmp.ttDBERFvDc source...        
      DEBU[2023-10-04T09:27:21Z] progress rsync /tmp/tmp.ttDBERFvDc/ to /run/cos/workingtree/: 80.04 / rem. 1703 / tot. 8530 / sp. 149.33kB/s 
      DEBU[2023-10-04T09:27:26Z] progress rsync /tmp/tmp.ttDBERFvDc/ to /run/cos/workingtree/: 86.82 / rem. 1124 / tot. 8530 / sp. 53.69kB/s 
      DEBU[2023-10-04T09:27:31Z] progress rsync /tmp/tmp.ttDBERFvDc/ to /run/cos/workingtree/: 89.70 / rem. 1328 / tot. 12887 / sp. 17.32kB/s 
      DEBU[2023-10-04T09:27:36Z] progress rsync /tmp/tmp.ttDBERFvDc/ to /run/cos/workingtree/: 92.82 / rem. 1083 / tot. 15086 / sp. 5.00kB/s 
      DEBU[2023-10-04T09:27:41Z] progress rsync /tmp/tmp.ttDBERFvDc/ to /run/cos/workingtree/: 94.01 / rem. 1177 / tot. 19650 / sp. 20.84kB/s 
      DEBU[2023-10-04T09:27:46Z] progress rsync /tmp/tmp.ttDBERFvDc/ to /run/cos/workingtree/: 94.91 / rem. 1204 / tot. 23645 / sp. 35.67kB/s 
      DEBU[2023-10-04T09:27:51Z] progress rsync /tmp/tmp.ttDBERFvDc/ to /run/cos/workingtree/: 95.99 / rem. 1027 / tot. 25592 / sp. 1.32kB/s 
      DEBU[2023-10-04T09:27:56Z] progress rsync /tmp/tmp.ttDBERFvDc/ to /run/cos/workingtree/: 95.14 / rem. 1310 / tot. 26950 / sp. 9.13kB/s 
      DEBU[2023-10-04T09:28:01Z] progress rsync /tmp/tmp.ttDBERFvDc/ to /run/cos/workingtree/: 96.69 / rem. 963 / tot. 29131 / sp. 0.00kB/s 
      INFO[2023-10-04T09:28:02Z] Finished copying /tmp/tmp.ttDBERFvDc into /run/cos/workingtree 
      ...


### Test Information
* Test Environment: Single nodes harvester on local kvm machine
* Harvester version: v1.2-04b7f9d7 (10/03)

### Verify Steps
<details><summary>Execution Steps</summary>

  1. Prepare a single or multiple nodes v1.1.2 Harvester
  2. Upgrade to latest v1.2-head
  3. Monitoring the upgrade process 
  #### For single node
  1. ssh to the Harvester node machine
  2. After the upgrade complete
  3. Find the upgrade pod in harvester-system namespace 
    ```
    kubectl get pod -n harvester-system
    ```
  4. Check the logs of the single node upgrade pod 
    ```
    kubectl logs hvst-upgrade-sfgvp-single-node-upgrade-harv112upd-j27b2 -n harvester-system
    ```
  5. Check the logs of the single upgrade pod cotains `DEBU` label
  
  #### For multiple nodes 
  1. ssh to the Harvester node machine
  2. When the upgrade proceed to `post-drain` mode 
  3. Check the current upgrade jobs
    ```
    kubectl get job -A
    ```
  4. When the upgrade jobs contains `post-drain` 
  5.  Find the of the post-drain pod
    ```
    kubectl get pod --selector=harvesterhci.io/upgradeJobType=post-drain -n harvester-system
    ```
  6.  Check the log of the post-drain pod contains `DEBU` label
  

</details>
